vendor/knative.dev/operator/pkg/reconciler/knativeeventing/controller.go

```diff
@@ -59,10 +59,10 @@ func NewExtendedController(generator common.ExtensionGenerator) injection.Contro
 		c := &Reconciler{
 			kubeClientSet:     kubeClient,
 			operatorClientSet: operatorclient.Get(ctx),
-			extension:         generator(ctx),
 			manifest:          manifest,
 		}
 		impl := knereconciler.NewImpl(ctx, c)
+		c.extension = generator(ctx, impl)
```

Comment:
Probably unrelated (as it's a rebase to upstream, right ?), but this kind of cyclic dependency is typically an architectural smell. Not sure if this has to be like this or whether there is a good reason for this (one drawback of course is, that later refactoring gets much more difficult if those objects are weaved together so closely.